yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/DockerCloud.java

```diff
@@ -221,12 +221,13 @@ private DockerSlave provisionWithWait(DockerSlaveTemplate template, Provisioning
         final DockerContainerLifecycle dockerContainerLifecycle = template.getDockerContainerLifecycle();
         final String imageId = dockerContainerLifecycle.getImage();
         final DockerComputerLauncher computerLauncher = template.getLauncher();
+        final String osType = this.getClient().infoCmd().exec().getOsType();
```

Comment:
Have you checked a swarm? AFAIK the swarm will not return an osType. I did not know of the osType of an Image that can be obtained by inspecting it ... so that should be the way to go.